app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -1,331 +1,100 @@
+  // triggered by cookie-consent event, which happens when users consent to cookies
+  Ga4LinkTracker.prototype.startModule = function () {
+    if (window.dataLayer) {
+      this.handleClick = this.handleClick.bind(this)
+      this.handleMousedown = this.handleMousedown.bind(this)
 
-      if (element.tagName !== 'A') {
-        element = element.closest('a')
-      }
+      this.module.addEventListener('click', this.handleClick)
+      this.module.addEventListener('contextmenu', this.handleClick)
+      this.module.addEventListener('mousedown', this.handleMousedown)
+    }
+  }
 
-      if (!element) {
-        return
+  Ga4LinkTracker.prototype.handleClick = function (event) {
+    var target = event.target
+    if (!this.trackLinksOnly) {
+      this.trackClick(event)
+    } else if (this.trackLinksOnly && target.closest('a')) {
+      if (!this.limitToElementClass) {
+        this.trackClick(event)
+      } else if (target.closest('.' + this.limitToElementClass)) {
+        this.trackClick(event)
       }
+    }
+  }
 
-      var href = element.getAttribute('href')
+  Ga4LinkTracker.prototype.handleMousedown = function (event) {
+    // 1 = middle mouse button
+    if (event.button === 1) {
+      this.handleClick(event)
+    }
+  }
 
-      if (!href) {
+  Ga4LinkTracker.prototype.trackClick = function (event) {
+    var target = window.GOVUK.analyticsGa4.core.trackFunctions.findTrackingAttributes(event.target, this.trackingTrigger)
+    if (target) {
+      var schema = new window.GOVUK.analyticsGa4.Schemas().eventSchema()
+
+      try {
+        var data = target.getAttribute(this.trackingTrigger)
+        data = JSON.parse(data)
+      } catch (e) {
+        // if there's a problem with the config, don't start the tracker
+        console.error('GA4 configuration error: ' + e.message, window.location)
         return
       }
-      var clickData = {}
-      var linkAttributes = element.getAttribute('data-ga4-link')
-      if (linkAttributes) {
-        clickData = JSON.parse(linkAttributes)
 
-        /* Since external links can't be determined in the template, we use populated-via-js as a signal
-        for our JavaScript to determine this value. */
-        if (clickData.external === 'populated-via-js' && clickData.url) {
-          clickData.external = this.isExternalLink(clickData.url) ? 'true' : 'false'
-        }
+      schema.event = 'event_data'
+      data.text = window.GOVUK.analyticsGa4.core.trackFunctions.removeLinesAndExtraSpaces(event.target.textContent)
+      data.url = window.GOVUK.analyticsGa4.core.trackFunctions.removeCrossDomainParams(this.findLink(event.target).getAttribute('href'))
+      data.link_path_parts = window.GOVUK.analyticsGa4.core.trackFunctions.populateLinkPathParts(data.url)
+      data.method = window.GOVUK.analyticsGa4.core.trackFunctions.getClickType(event)
+      data.external = window.GOVUK.analyticsGa4.core.trackFunctions.isExternalLink(data.url, []) ? 'true' : 'false'
```

Comment:
See other comment about ensuring the current hostname is in the array you pass.

Reply:
Changed.